Feature/decorate class

RileyManda opened the pull request:
I have completed the required implementation of the School Library Application according to the stipulated [requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md):

The following features were implemented:

### Decorator Classes

- [x] Created class nameable|converted person class to nameable
- [x] Implemented nameable and adding nameable inheritance to person class 
- [x] Implemented capitalize decorator
- [x] Implemented trimmer decorator class and decorated person class with test_file data


In addition, I have fulfilled the following project requirements:

### General requirements

- There are [no linter errors](https://github.com/microverseinc/linters-config).
- Used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- Documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
- Followed the list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).



Please kindly review my implementation and suggest any changes and improvements.

I look forward to your feedback.

Thank you 🙏🏻 
 Best regards, 

RileyManda committed:
Created class nameable|turned person class to nameable

RileyManda committed:
Implemented nameable and adding nameable inheritance to person class

RileyManda committed:
Implemented capitalize decorator

RileyManda committed:
Implemented trimmer decorator class and tested the person class with test_file data

RileyManda committed:
Linter and syntax fixes

topeogunleye commented on person.rb:
```diff
@@ -1,3 +1,4 @@
+require_relative 'nameable'
 class Person
```
Great job on implementing `require_relative 'nameable'` in this file, however, I have some suggestions for improvement:

- 🚨 You have not inherited from the `Nameable` class in your `Person` class. This means that your `Person` class does not have access to the methods and attributes of the `Nameable` class. [This violates the requirement of inheritance in Ruby](https://www.rubyguides.com/2019/01/what-is-inheritance-in-ruby/)¹². To fix this, you need to use the `<` operator to indicate that your `Person` class is a subclass of the `Nameable` class. For example, you should write:

```ruby
class Person < Nameable
  # your code here
end
```


I hope this feedback helps you improve your code and your coding skills. Please let me know if you have any questions or need further clarification. Happy coding! 😊

RileyManda replied:
Thank you for the review and feedback @topeogunleye 
I will make sure to implement the suggested changes.
Thank you 🙏🏻

topeogunleye replied:
you're welcome @RileyManda. Please kindly submit for review

RileyManda replied:
Hi @topeogunleye 
I have submitted for a second review.Thank you

topeogunleye reviewed (changes requested):
Hi @RileyManda, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

![Try-Again](https://user-images.githubusercontent.com/52934128/180646051-aed7efa6-6d15-4853-9962-73186e366f2c.gif)

> ## (Highlights) Good Points: 💪 

- 👍 You have defined the Nameable class with an abstract correct_name method that raises a NotImplementedError. This is a good practice to define an interface that other classes must implement.
- 👍 You have defined the Decorator class that inherits from the Nameable class and has an attribute nameable that stores a namable object. You have also implemented the correct_name method that delegates to the correct_name method of the nameable object. This is a good way to implement the base decorator class that wraps another object and adds some functionality.
- 👍 You have defined two subclasses of the Decorator class: CapitalizeDecorator and TrimmerDecorator. These classes override the correct_name method and add some logic to modify the name of the namable object. This is a good way to implement concrete decorators that provide specific behaviors.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Nothing to mention.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

RileyManda committed:
Implemented super and relative:nameable in person class

RileyManda commented:
I have completed implementing the suggested changes.

Changes made:

- [x] Added require relative nameable to person class(Inheritance)
- [x] Implemented super to resolve missing super in the constructor method.

Please kindly review my implementation and suggest any changes and improvements.

I look forward to your feedback.

Thank you 🙏🏻
 Best regards, 

gemmen29 reviewed (approved):
Hi @RileyManda, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

![](https://media.giphy.com/media/mCIjCgs3nWQWfJZvPA/giphy-downsized-large.gif)

## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
